src/auxiliary/math.jl

```diff
@@ -7,6 +7,21 @@
 
 const TRIXI_UUID = UUID("a7f1ee26-1774-49b1-8366-f1abc58fbfcb")
 
+"""
+    Trixi.set_polyester(toggle::Bool; force = true)
+
+Toggle the usage of [Polyester.jl](https://github.com/JuliaSIMD/Polyester.jl) for multithreading.
+By default, Polyester.jl is enabled, but it can
+be useful for performance comparisons to switch to the Julia core backend.
+
+This does not fully disable Polyester.jl,
+buy only its use as part of Trixi.jl's `@threaded` macro.
+"""
+function set_polyester(toggle::Bool; force = true)
```

Comment:
```suggestion
function set_polyester!(toggle::Bool; force = true)
```
If `set_preferences!` has a `!`, shouldn't `set_polyester` also get one? (I just see that `set_sqrt_type` also doesn't have a `!`, but, e.g., `set_library_p4est!` from P4est.jl has one.)